observer/observer.go

```diff
@@ -152,14 +158,57 @@ func (o *Observer) Watch(files []string) error {
 	return nil
 }
 
-// handleEvent handle an event.
-func (o *Observer) handleEvent(event interface{}) {
-	// Run all listeners for this event.
+// SetBufferDuration set the buffer damping duration
+func (o *Observer) SetBufferDuration(d time.Duration) {
+	// Create the buffer mutex, if missing.
+	if o.bufferMux == nil {
+		o.bufferMux = &sync.Mutex{}
+	}
+
+	// Set the buffer duration.
+	o.bufferDuration = d
+}
+
+// sendEvent send one or more events to the observer listeners.
+func (o *Observer) sendEvent(event interface{}) {
 	for _, listener := range o.listeners {
 		go listener(event)
 	}
 }
 
+// handleEvent handle an event.
+func (o *Observer) handleEvent(event interface{}) {
+	// If we do not buffer events, just send this event now.
+	if o.bufferDuration == 0 {
+		o.sendEvent(event)
+		return
+	}
+
+	// Add new event to the event buffer
+	o.bufferMux.Lock()
+	o.bufferEvents = append(o.bufferEvents, event)
+	o.bufferMux.Unlock()
+
+	// If this is the first event, set a timeout function.
+	if o.bufferTimer == nil {
+		o.bufferTimer = time.AfterFunc(o.bufferDuration, func() {
+			// Run all listeners for this event.
+			o.bufferMux.Lock()
+
+			// Send event buffer
+			o.sendEvent(o.bufferEvents)
```

Comment:
That's really interesting: `o.sendEvent` takes an interface while here you are passing a list of interfaces.
I guess it compiles because the `go` compiler treats `interface{}` as blob so passing `[]interface{}` will be considered as `interface{}` which may send an awkward event.